crates/cargo-test-support/src/paths.rs

```diff
@@ -118,6 +145,23 @@ pub fn root() -> PathBuf {
     root
 }
 
+/// Path to the test's filesystem scratchpad
+///
+/// ex: `$CARGO_TARGET_TMPDIR/cit/t0`
+#[cfg(not(windows))]
+pub fn root() -> PathBuf {
+    let test_name = TEST_NAME.with(|n| {
+        n.borrow().clone().expect(
+            "Tests must use the `#[cargo_test]` attribute in \
+             order to be able to use the crate root.",
+        )
+    });
+
+    let mut root = global_root();
+    root.push(&test_name);
+    root
+}
+
```

Comment:
Part 5 of removing the platform-specific code.

This also updates the doc-comment to mention both forms.

```suggestion
/// Path to the test's filesystem scratchpad
///
/// ex: `$CARGO_TARGET_TMPDIR/cit/<integration test>/<module>/<fn name>/`
/// or `$CARGO_TARGET_TMPDIR/cit/t0` on Windows
pub fn root() -> PathBuf {
    let test_dir = TEST_DIR.with(|n| {
        n.borrow().clone().expect(
            "Tests must use the `#[cargo_test]` attribute in \
             order to be able to use the crate root.",
        )
    });

    let mut root = global_root();
    root.push(&test_dir);
    root
}

```